src/Tcp.roc

```diff
@@ -0,0 +1,110 @@
+interface Tcp
+    exposes [
+        Stream,
+        withConnect,
+        readBytes,
+        readUtf8,
+        writeBytes,
+        writeUtf8,
+    ]
+    imports [Effect, Task.{ Task }, InternalTask, InternalTcp]
+
+Stream : InternalTcp.Stream
+
+ConnectErr : InternalTcp.ConnectErr
+
+StreamErr : InternalTcp.StreamErr
+
+## Opens a TCP conenction to a remote host and perform a [Task] with it.
+##
+##     # Connect to localhost:8080 and send "Hi from Roc!"
+##     stream <- Tcp.withConnect "localhost" 8080
+##     Tcp.writeUtf8 "Hi from Roc!" stream
+##
+## Examples of valid hostnames:
+##  - 127.0.0.1
+##  - ::1
+##  - localhost
+##  - roc-lang.org
+##
+## The connection is automatically closed after the [Task] is completed.
+withConnect : Str, U16, (Stream -> Task {} err) -> Task {} [TcpConnectErr ConnectErr, TcpPerformErr err]
+withConnect = \hostname, port, callback ->
+    stream <- connect hostname port
+        |> Task.mapFail TcpConnectErr
+        |> Task.await
+
+    {} <- callback stream
+        |> Task.mapFail TcpPerformErr
+        |> Task.await
+
+    close stream
+
+connect : Str, U16 -> Task Stream ConnectErr
+connect = \host, port ->
+    Effect.tcpConnect host port
+    |> Effect.map InternalTcp.fromConnectResult
+    |> InternalTask.fromEffect
+
+close : Stream -> Task {} *
+close = \stream ->
+    Effect.tcpClose stream
+    |> Effect.map \_ -> Ok {}
+    |> InternalTask.fromEffect
+
+## Reads all available bytes in the TCP Stream.
```

Comment:
The read functions try to read everything available. This may be desirable since this is a _basic_ CLI platform. 

However, if you're implementing a protocol, it's common to know how much you need to read.

Should we also expose a `readUpTo` function?

Reply:
I am +1 on supporting such a thing. I would suggest "readExact" that blocks on reading N bytes, or errors if EOF is reached. This will make it simple to implement most network message protocols without having to loop.

Reply:
Yeah, great point. I'll add this!

Reply:
@ayazhafiz Added all-new reading functions!